bodhi/server/consumers/pungi_wrapper.py

```diff
@@ -0,0 +1,498 @@
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The Bodhi "Masher".
+
+This module is responsible for the process of "pushing" updates out. It's
+comprised of a fedmsg consumer that launches threads for each repository being
+mashed.
+"""
+
+import os
+import sys
+import getpass
+import cStringIO
```

Comment:
When we switch Bodhi to Python 3 I don't think we'll need to support Python 2 anymore, so I think leaving the use of ```cStringIO``` here for now will be OK since we won't need to support Python 2 and 3 at the same time.